development-docs/TESTING.md

```diff
@@ -0,0 +1,188 @@
+ 
+- `Utils` static methods
+- `Clients` clients for testing overall communication of the `Kafka Bridge`
+- `Facades` encapsulation of the standalone `Kafka` and `AdminClient` instance
+
+Furthermore we divide our test suite on two separate parts:
+
+1. `Http`package, where you can find all test cases related to the `Http Kafka Bridge`
+2. `Amqp`package, where you can find all tests cases related to the `Amqp Kafka Bridge`
+
+## Approaches
+
+There are four approaches how you can verify that `Kafka Bridge` is without any bug. Before I list all approaches 
+the following  terminology is needed.
+
+#### In-memory 
+`Kafka` = 
+- for `Http Kafka Bridge` we use [Strimzi Kafka container]([here](https://github.com/strimzi/strimzi-kafka-operator/tree/master/test-container).)
+- for `Amqp Kafka Bridge` we use [KafkaFacade](https://github.com/strimzi/strimzi-kafka-bridge/blob/master/src/test/java/io/strimzi/kafka/bridge/facades/KafkaFacade.java) 
+
+`Kafka Bridge` =
+ - for deployment of the `Http Kafka Bridge` and `Amqp Kafka Bridge` we are using [Vert.x](https://vertx.io/) 
+
+#### Standalone 
+`Kafka` 
+ - we use latest version of the `Kafka` and you can start the `Kafka` instance by the `zookeeper-server-start.sh` and
+`kafka-server-start.sh`, which can be for instance downloaded from [this](https://kafka.apache.org/downloads) site
+
+`Kafka Bridge` 
+ - start `Kafka Bridge` instance using bash script located `target/kafka-bridge-{version}/kafka-bridge-{version}/bin/kafka_bridge_run.sh` 
+with following command `kafka_bridge_run.sh --config-file ../config/application.properties`
+ 
+----  
+The approaches are as follows:
+
+|  Approach	| Supported |
+|---	|---	| 
+|   1. Using in-memory `Kafka` and in-memory `Kafka Bridge` | Yes
+|   2. Using standalone `Kafka` and standalone `Kafka Bridge` | Yes
+|   3. Using in-memory `Kafka` and standalone `Kafka Bridge` | No
+|   4. Using standalone `Kafka` and in-memory `Kafka Bridge` | No
+
+In our CI we are currently testing the first approach which is the most important. Also, in the future we are planning 
+to add the second option. Rest of these approaches are not so essential. 
+
+## Test Phases
+
+In general, we use classic test phases: `setup`, `exercise`, `verify` and `teardown`. Every phase will be described for the
+`src.test.java.io.strimzi.kafka.bridge.http` package. It is very similar with `src.test.java.io.strimzi.kafka.bridge.amqp` only
+one difference is that `http` package using Strimzi `Kafka` container by `test-containers` and `amqp` package using KafkaFacade.
+
+### Setup
+
+In this phase we perform:
+
+* Deploy the in-memory Strimzi `Kafka` container or standalone `Kafka`
+* Deploy the AdminClient
+* Deploy the in-memory `Kafka Bridge` or standalone `Kafka-Bridge`
+* Deploy the WebClient
+
+Everything is setup in the `HttpBridgeTestBase` class. In the static block we instantiate the Strimzi `Kafka` container,
```

Comment:
Maybe we should rename `HttBridgeTestBase`to `HttpBridgeTestAbstract` since we use Abstrac naming in Strimzi as well. But it's just a note.